parts/k8s/kubernetesmastercustomdata.yml

```diff
@@ -73,7 +73,11 @@ write_files:
     - name: localcluster
       cluster:
         certificate-authority: /etc/kubernetes/certs/ca.crt
-        server: {{WrapAsVerbatim "concat('https://', variables('masterPrivateIpAddrs')[copyIndex(variables('masterOffset'))], ':443')"}}
+        {{- if IsPrivateCluster}}
+            server: {{WrapAsVariable "kubeconfigServer"}}
+        {{- else}}
+            server: {{WrapAsVerbatim "concat('https://', variables('masterPrivateIpAddrs')[copyIndex(variables('masterOffset'))], ':443')"}}
```

Comment:
@lachie83 now I'm wondering if we need this if statement... Since the server IP was already the master's IP address and it's different for each master I don't think changing it to the ILB IP for >1 master / the first master IP for 1 master makes that much sense. What do you think?

Reply:
Yeah I agree. This does however raise a question around the failure-modes if kubelet can only communicate with the local api-server in a multi-master cluster.